### PR TITLE
kafka: upstream kafka facade in mesh-filter

### PR DESCRIPTION
**Pull request by adamkotwasinski**
Commit Message: upstream kafka facade in mesh-filter
Additional Description: Thread-local container/manager for Kafka producers that will be accessed by Kafka produce requests [like this](https://github.com/envoyproxy/envoy/blob/a59244b08803c4bf5a152ae7804cc57cc1082921/contrib/kafka/filters/network/source/mesh/command_handlers/produce.cc#L29). Only facade part, the Kafka producer wrapper (together with librdkafka dependency) will be in another PR.
Risk Level: Low
Testing: Unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: contrib/kafka/filters/network/source/mesh/upstream_kafka_client_impl.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+namespace Mesh {
+
+using RawKafkaProducerConfig = std::map<std::string, std::string>;
+
+// Placeholder for proper Kafka Producer object.
+// It will also keep a reference to a dedicated thread (that's why we need a factory) that's going
+// to be polling for delivery notifications.
```
Comment (adamkotwasinski) [inline contrib/kafka/filters/network/source/mesh/upstream_kafka_client_impl.h:17]: [relevant real code](https://github.com/envoyproxy/envoy/blob/a59244b08803c4bf5a152ae7804cc57cc1082921/contrib/kafka/filters/network/source/mesh/upstream_kafka_client.cc#L60)

**Comment by mattklein123** [comment]
Please check CI.

/wait

**Comment by adamkotwasinski** [comment]
Looks like was some fluke in docs generation, went away with retest

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**